src/main/java/com/example/onlinebookstore/service/order/impl/OrderServiceImpl.java

```
@@ -0,0 +1,147 @@
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final UserService userService;
+
+    @Override
+    public List<OrderResponseDto> getOrdersForCurrentUser(Pageable pageable) {
+        return orderRepository.getOrdersByUser(userService.getCurrentUser(), pageable)
+                .stream()
+                .map(this::convertToDtoOrderAndSetItemsDto)
+                .toList();
+    }
+
+    @Override
+    public OrderResponseDto createOrder(OrderRequestDto request) {
+        ShoppingCart cart = getShoppingCartByUser(userService.getCurrentUser());
+        Order order = new Order();
+        order.setUser(userService.getCurrentUser());
+        cart.getCartItems().stream()
+                .map(orderItemMapper::convertCartItemToOrderItem)
+                .forEach(item -> item.addOrder(order));
+        order.setTotal(getTotal(order));
+        order.setShippingAddress(request.getShippingAddress());
+        orderRepository.save(order);
+        orderItemRepository.saveAll(order.getOrderItems());
+        return convertToDtoOrderAndSetItemsDto(order);
+    }
+
+    @Override
+    public void updateOrderStatus(Long id, OrderStatusDto statusDto) {
+        Optional<Order> orderOptional = orderRepository.findById(id);
+        if (orderOptional.isPresent()) {
+            Order order = orderOptional.get();
+            order.setStatus(statusDto.getStatus());
+            orderRepository.save(order);
+        } else {
+            throw new EntityNotFoundException("There is no Order with id: " + id);
+        }
```

```suggestion
        Order order = orderRepository.findById(id)
                .orElseThrow(() -> new EntityNotFoundException("There is no Order with id: " + id));
    
        order.setStatus(statusDto.getStatus());
        orderRepository.save(order);
```
fail early where possible